src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -798,6 +798,13 @@ bool SHAMap::addItem (const SHAMapItem& i, bool isTransaction, bool hasMetaData)
     return addGiveItem(std::make_shared<SHAMapItem const>(i), isTransaction, hasMetaData);
 }
 
+bool
+SHAMap::addItem(SHAMapItem&& i, bool isTransaction, bool hasMetaData)
+{
+    return addGiveItem(std::make_shared<SHAMapItem const>(std::move(i)),
```

Comment:
Love this!

Quibble: This is a weird read with the code at the far right, and Vinnie likes to get everything all foveal.  :-)  

Perhaps have the carriage return before `std::move()`?  Consider calling the variable `item` as `i` is 99% of the time an iterator or loop index.